### DISCUSSION TITLE
SuperSimpleNet does not work on datasets without groundtruth masks

### DISCUSSION THREAD

**Discussion by mzweilin** [open-edge-platform/anomalib#2552] [Feature Requests]
Hello @blaz-r , thanks for implementing SuperSimpleNet. I tried to train a SuperSimpleNet model, but it doesn't work because my dataset does not have mask annotations. I believe the mask annotation is not required in the unsupervised mode. Would you be able to fix it? Thanks.

**Comment by blaz-r** [comment]
Hi. Anomalib dataset should contain the mask even if it's not actually present in the dataset:
https://github.com/openvinotoolkit/anomalib/blob/d437f231ddfcb95fac75eecd3e032d3cbb7ca28e/src/anomalib/data/datasets/base/image.py#L278-L282

Are you using a custom non-anomalib dataset? 
In that case, a workaround would be to create an empty mask in `training_step` or `forward` pass. I think I could open a PR for this, which would also make the model usable for any dataset.

**Reply by mzweilin** [reply]
Hi Blaž, thanks for your reply. My dataset does not have the pixel-level annotations, so I have to load them as `TaskType.CLASSIFICATION`, not `TaskType.SEGMENTATION`. The datamodule does not add the empty mask in this case.

Yes, it will be great if your model supports classfication datasets!

**Reply by blaz-r** [reply]
So a very important thing is that you please use this only if your problem is unsupervised. The code is currently not made to work for weakly labelled (image-level only) anomalous images. So if you have an unsupervised classification problem, you can fix this issue by adding this inside forwards pass training section:
```python
            if masks is None:
                b, _, h, w = features.shape
                masks = torch.zeros((b, 1, h, w), dtype=torch.float32, device=features.device)
            elif masks is not None and labels is not None:
                masks = self.downsample_mask(masks, *features.shape[-2:])
```
I would rather avoid making a PR for this, since it could lead to unexpected behaviour in case of weakly labelled anomalous data present. We are working on a version that explicitly supports that as well, but until that is released, you can use the code above, but as I said only for unsupervised case.
